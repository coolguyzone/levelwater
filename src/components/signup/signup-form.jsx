import React, { Component } from 'react';
import { Link, Route } from 'react-router-dom';
import { Field, reduxForm } from 'redux-form';
import { signUserUp } from '../../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import styles from '../login/login.css';

const mapStateToProps = (state, ownProps) => {
  return {
    signupFormData: state.form
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators ({ signUserUp }, dispatch);
};

export class SignupForm extends Component{
  render(props) {
    const { handleSubmit, pristine, reset, submitting } = this.props;
    return (
      <Route render={({ history }) => (
      <form id="signup-form" onSubmit={(event) => {
        event.preventDefault();
        let x = this.props.signupFormData.signup.values;
        this.props.signUserUp(x.email, x.firstName, x.lastName, x.password, x.passwordConfirm, () => {
          history.push('/signup/step1');
        });
      }}>
      <Link to="/"><div id="close-login"><img
    src={require('../../img/black-icon.png')}
     alt="close window"
   /></div></Link>
        <div className="ui grid">

          <div className="login-field signup-field column sixteen wide">

            <div>
              <Field name="email" component="input" type="email" placeholder="Email" required />
            </div>
            <div>
              <Field name="firstName" component="input" type="text" placeholder="First Name" required />
            </div>
            <div>
              <Field name="lastName" component="input" type="text" placeholder="Last Name" required />
            </div>
            <div>
              <Field name="password" component="input" type="password" placeholder="Password" required />
            </div>
            <div>
              <Field name="passwordConfirm" component="input" type="password" placeholder="Confirm Password" required />
            </div>
          </div>

          <div className="login-field column sixteen wide">
            <button type="submit">Sign Up</button>
          </div>
        </div>
        <div id="login-border"></div>
      </form>
      )} />
    );
  }
}

SignupForm = reduxForm ({
  form: 'signup'
})(SignupForm);



export default connect (mapStateToProps, mapDispatchToProps)(SignupForm);
