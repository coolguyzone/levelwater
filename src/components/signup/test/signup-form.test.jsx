import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SignupForm } from '../signup-form.jsx';


describe('signup form', () => {
  it('renders a snapshot', () => {
    const tree = shallow(<SignupForm />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
