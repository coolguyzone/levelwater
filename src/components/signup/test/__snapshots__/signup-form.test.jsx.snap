exports[`signup form renders a snapshot 1`] = `
<Connect(Form(SignupForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="signup"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false} />
`;
